x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/single_bulk_create.ts

```diff
@@ -43,6 +65,8 @@ export const singleBulkCreate = async ({
   enabled,
   tags,
 }: SingleBulkCreateParams): Promise<boolean> => {
+  someResult.hits.hits = filterDuplicateRules(id, someResult);
```

Comment:
It'd be nice to only have to run this filter when we know we are searching the signals index, and not some other index. Can we pass in the "searched" index as a parameter to this function and check if the "searched" index equals the signals index. Then we would know that we are processing signals-on-signals and can then know when to run this filter, or skip the iteration all-together when we know we're not doing signals-on-signals.

Reply:
Update: Can't guarantee that the input signals index is the same as the output signals index. Therefore we are not able to determine when we might be doing signals on signals during bulk create. Also it's not a major performance issue at the moment and there are other ways to mitigate any issues that might arise before this filter becomes a problem.